lib/ohai/plugins/virtualbox.rb

```diff
@@ -20,8 +22,113 @@
   depends "virtualization"
   provides "virtualbox"
 
+  # query virtualbox for each configured vm, as well as
+  # each guest's individual configuration settings
+  def vboxmanage_list_vms
+    vms = Mash.new
+    so_cmd = "VBoxManage list --sorted vms"
+    logger.trace(so_cmd)
+    so = shell_out(so_cmd)
+
+    if so.exitstatus == 0
+      # parse the output
+      so.stdout.lines.each do |line|
+        case line
+        when /^"(\S*)" \{(\S*)\}$/
+          name = Regexp.last_match(1)
+          uuid = Regexp.last_match(2)
+          vms[name] = vboxmanage_vminfo(uuid)
+        end
+      end
+    end
+    vms
+  rescue Ohai::Exceptions::Exec
+    logger.trace("Plugin VboxHost: Could not run 'VBoxManage list --sorted vms'. Skipping data")
+  end
+
+  # query the vminfo for particular guest instance, normalizing
+  # the fields so that they're not enclosed in double-quotes (")
+  def vboxmanage_vminfo(machine_id)
+    vm = Mash.new
+
+    so_cmd = "VBoxManage showvminfo #{machine_id} --machinereadable"
+    logger.trace(so_cmd)
+    so = shell_out(so_cmd)
+
+    if so.exitstatus == 0
+      so.stdout.lines.each do |line|
+        line.chomp!
+        left, right = line.split("=")
+
+        # remove enclosing quotes, if needed
+        key = left.delete_prefix('"').delete_suffix('"')
+
+        # skip the name attribute since that is the parent key
+        next if left == "name"
```

Comment:
Maybe want to test `key` here after the `"` chomping on `left`. Is the value always lowercase `name` or should this test be after downcasing?

Reply:
Updated